### PR TITLE
AM 1.7 upgrades

### PR DESCRIPTION
**Pull request by sevein**
This needs #681 and #682 (AM) and https://github.com/artefactual/archivematica-storage-service/pull/217 (SS).

It includes the changes that Helen did in #120.

I'd also like to address #678 here but can be done in a separate PR.

Reminders (once merged):
- File issue about problems with upgrade path
- Update deploy-pub and getting started docs
- Document environment variables
- Support automated user creation

### CODE REVIEW

**Comment by scollazo** [comment]
I think that keeping the .j2 extension for templates is a good idea, so we are aware that they support jinja2 templates inside.

**Comment by helenst** [comment]
I pushed a small fix to this branch, as the provisioner was failing trying to locate the upstart script for storage service.

File: defaults/main.yml [hunk 1]
```diff
@@ -1,162 +1,117 @@
...
+  DJANGO_SETTINGS_MODULE: "storage_service.settings.production"
+  DJANGO_ALLOWED_HOSTS: "*"
+  DJANGO_SECRET_KEY: "CHANGE_ME_WITH_A_SECRET_KEY"
+  SS_DB_HOST: ""
+  SS_DB_NAME: "/var/archivematica/storage-service/storage.db"  # Needed by ss-db.yml
+  SS_DB_PASSWORD: ""
+  SS_DB_USER: ""
+  # Not mandatory:
+  SS_GUNICORN_BIND: "127.0.0.1:8002"
+
```
Comment (helenst) [inline defaults/main.yml]: I notice AM / SS back end ports are swapped from what they were previously 8001 <-> 8002. Makes sense to me I guess, to have the AM port 'first'. Although caused some confusion as I sometimes kill the gunicorn service and replace it with a runserver for easier debugging! Is there anywhere these ports are documented that will need to be changed?

Comment (sevein) [reply defaults/main.yml]: I didn't mea to change them, that was an accident! I'll rever that. Thank you!

**Comment by jrwdunham** [comment]
In order to get this to work, I had to insert `archivematica_src_am_wait_mysql: "yes"` into vars-singlenode-qa.yml (using Trusty). Should something like this be in defaults/main.yml?

**Comment by sevein** [comment]
> In order to get this to work, I had to insert archivematica_src_am_wait_mysql: "yes" into vars-singlenode-qa.yml (using Trusty). Should something like this be in defaults/main.yml?

That was a typo, fixed now, thanks @jrwdunham.

**Comment by helenst** [comment]
I've been using this all day and it seems to be working well now :+1:

**Review by jhsimpson — APPROVED** [review]
I tested this on windows, using @minusdavid's deploy-pub pr https://github.com/artefactual/deploy-pub/pull/39
 works good for me
👍

**Pull request merged**